torch_geometric/graphgym/model_builder.py

```diff
@@ -17,14 +25,37 @@
 
 
 class GraphGymModule(LightningModule):
-    def __init__(self, dim_in, dim_out, cfg):
+    def __init__(self, dim_in, dim_out, cfg: "CfgNode"):
         super().__init__()
+        self.cfg = cfg
         self.model = network_dict[cfg.model.type](dim_in=dim_in,
                                                   dim_out=dim_out)
 
     def forward(self, *args, **kwargs):
         return self.model(*args, **kwargs)
 
+    def configure_optimizers(self) -> Tuple[Any, Any]:
+        optimizer = create_optimizer(self.model.parameters(), self.cfg.optim)
+        scheduler = create_scheduler(optimizer, self.cfg.optim)
+        return [optimizer], [scheduler]
+
+    def _shared_step(self, batch, split: str) -> Dict:
+        batch.split = split
```

Comment:
Any reason we do this?

Reply:
I referred to the [current training implementation](https://github.com/aniketmaurya/pytorch_geometric/blob/implement_module/torch_geometric/graphgym/train.py#L45) for this.

Reply:
Ah, thanks for clarifying.